src/components/navigation/header/navigationTenantMenu/index.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC, useCallback, useMemo, useRef, useState } from 'react';
+import { useI18n } from '@smg-automotive/i18n-pkg';
+import { EnrichedSessionUser } from '@smg-automotive/auth';
+import {
+  Button,
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  Portal,
+  useDisclosure,
+} from '@chakra-ui/react';
+
+import Text from 'src/components/text';
+import { TenantSelectionSelectList } from 'src/components/tenantSelection/select/List';
+import { ChevronDownSmallIcon, GarageIcon } from 'src/components/icons';
+import Box from 'src/components/box';
+
+import { NavigationTenantMenuLoading } from './Loading';
+
+type Props = {
+  user: EnrichedSessionUser | null;
+  selectTenant: (sellerId: number | string) => Promise<void>;
+};
+
+const NavigationTenantMenu: FC<Props> = ({ user, selectTenant }) => {
+  const initialFocusRef = useRef<HTMLInputElement>(null);
+  const { onClose, isOpen, onToggle } = useDisclosure();
+  const [isLoading, setIsLoading] = useState(false);
+  const { t } = useI18n();
+  const onTenantSelect = useCallback(
+    async (selectedTenantId: number) => {
+      setIsLoading(true);
+      await selectTenant(selectedTenantId);
+      onClose();
+      setIsLoading(false);
+    },
+    [selectTenant, onClose],
+  );
+
+  const selectedTenant = useMemo(() => {
+    return user?.managedSellers?.find(
+      (seller) => seller.id === Number(user.sellerId),
+    );
+  }, [user?.managedSellers, user?.sellerId]);
+
+  if (!user || !user.isMultiTenantUser || !selectedTenant) return null;
+
+  return (
+    <Popover
+      placement="bottom-end"
+      returnFocusOnClose={true}
+      onClose={onClose}
+      isOpen={isOpen}
+      initialFocusRef={initialFocusRef}
+    >
+      <PopoverTrigger>
+        <Button
+          p="0"
+          color={isOpen ? 'blue.700' : 'gray.900'}
+          _hover={{ color: 'blue.700' }}
+          leftIcon={<GarageIcon />}
+          rightIcon={
+            <ChevronDownSmallIcon
+              transition="0.2s"
+              transform={isOpen ? 'rotate(180deg)' : 'rotate(0deg)'}
+            />
+          }
+          iconSpacing="xs"
+          onClick={onToggle}
+        >
+          <Text
+            as="span"
+            fontWeight="bold"
+            noOfLines={1}
+            maxW="2xl"
+            textAlign="left"
+          >
+            {selectedTenant.billingName || selectedTenant.id}
+            {selectedTenant.billingCity ? ', ' : null}
+            {selectedTenant.billingCity}
+          </Text>
```

Comment:
Wondering if we shuold use the tenat lable helper in here and make it consitent with the select

Reply:
Good point. It wasn't clear from the design what exactly we want to show here. I'll add it to the list of questions for Bojana, but makes sense keeping the two aligned